### PR TITLE
Sql standards draft

### PR DESCRIPTION
**Pull request by vtovar**
@dubgb @normrider2001 
Could you please review this Initial draft for SQL Standards? (I don't know how it looks with the format)

### CODE REVIEW

**Comment by dubgb** [comment]
I would propose indexes are prefixed with "idx_" instead of "ix_" if we want to match what has been largely followed already in the databases.

**Comment by vtovar** [comment]
I just replaced ix with idx.
Also I updated GasBuddyWiki with the same change. http://wiki.gbnetregina.local/wiki/1179/gblibrary

File: sql/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+Warning	891		SQL71558: The object reference [dbo].[MSA].[id] differs only by case from the object definition [dbo].[MSA].[ID].	C:\GasBuddyDBs\gb-sql-server-db\GEO\dbo\Stored Procedures\usp_gasbuddy_get_competitors_by_area_and_smid.sql	6	1	GEO
+
+```
+	SELECT		@areaGeog = geog
+	FROM		[dbo].[MSA] with (nolock)
+	WHERE		id = @areaID;
+```
+
+Fix: Since ID was definned with upper case in MSA table:
```
Comment (djMax) [inline sql/README.md]: [sp] defined

File: sql/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	WHERE id = @id;
+
+END
+
+```
+
+When joining, use aliases for the tables and identify all the columns with their corresponding alias.
+
+```
+	SELECT		srdt.sm_id ,
```
Comment (djMax) [inline sql/README.md]: Is this using tabs over spaces? Might be worth saying all indenting to be done with spaces?

File: sql/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+[dbo].[awards]
+[dbo].[members]
+[dbo].[member_comments]
+```
+
+
+formatting
+=========
+All T-SQL reserved words are uppercase.
+All identifier should be delimited by square brackets.
```
Comment (djMax) [inline sql/README.md]: [sp] identifiers

File: sql/README.md (hunk 2 above)
Comment (vtovar) [inline sql/README.md]: Let me confirm that.. I think that we are going to use tabs.

File: sql/README.md (hunk 1 above)
Comment (vtovar) [inline sql/README.md]: Fixed

File: sql/README.md (hunk 3 above)
Comment (vtovar) [inline sql/README.md]: Fixed

File: sql/README.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+	SELECT id
+		  ,display_name
+		  ,fuel_group_id
+		  ,octane_id
+		  ,ethanol_id
+		  ,category_id
+	FROM [dbo].[fuel_products] WITH(NOLOCK)
+	WHERE id = @id;
+
```
Comment (normrider2001) [inline sql/README.md:82]: The spacing example in this one should be the same as what is below.

File: sql/README.md (hunk 2 above)
Comment (normrider2001) [inline sql/README.md]: I say tabs, but this could start a war.

Comment (djMax) [inline sql/README.md]: I used to be a firm tabs guy, but after a long discussion with Griffin, I'm now a firm spaces guy. For me it's mostly about appearance, and as you can see from this diff, tabs are a mess in most cases.

File: sql/README.md (hunk 4 above)
Comment (vtovar) [inline sql/README.md:82]: Fixed

**Pull request merged**